Assets/Scripts/PlayerMovement.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+/// <summary>
+/// Controls the movement of a player object. Adds gravity and handles move inputs.
+/// </summary>
+[RequireComponent(typeof(CharacterController))]
+public class PlayerMovement : MonoBehaviour
+{
+    [SerializeField] private float moveSpeed;
+    [SerializeField] private float terminalVelocity;
+    private CharacterController cc;
+    private InputAction moveAction;
+    private Vector3 moveVector;
+
+    void Awake()
+    {
+        cc = GetComponent<CharacterController>();
+    }
+
+    void Start()
+    {
+        // returns Vector2 where x = left/right & y = up/down, on controller/keyboard
+        moveAction = InputSystem.actions.FindAction("Move");
+    }
+
+    void Update()
+    {
+        // handle walking movement (horizontal)
+        Vector2 moveInput = moveAction.ReadValue<Vector2>() * moveSpeed;
+        Vector2.ClampMagnitude(moveInput, moveSpeed);
```

Comment:
Good! This ensures that both straight and diagonal movement has identical movement speed. E.g., straight may have a speed of 1 and diagonal, being a combination of two straight movements, would have a movement speed of ~1.4. `√(1² x 1²) = 1.41...`